.github/workflows/CI_on_push.yml

```diff
@@ -0,0 +1,41 @@
+
+name: CI on push to main or issue-2 branch
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - issue-2-Setup_CI/CD_for_reference-models
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+
+      - name: Install clang and runtime dependencies
+        run: |
+          sudo apt install clang libunwind-dev
+          sudo apt install libgc-dev libre2-dev
+
+      - name: Checkout cli repo
+        uses: actions/checkout@v2
+        with:
+          repository: nervosum/nervosum
+          path: cli
+
+      - name: Build the CLI binary
+        run: |
+          cd cli
+          sbt cli/nativeLink
+
+      - name: Run CLI
+        run: cli/cli/target/scala-2.11/cli-out build
+
+
+
+
+
```

Comment:
Can you clean up whitespace before merging?

Reply:
Done!